chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -112,6 +113,8 @@ pub struct TransactionVerifier<C, S, U, A> {
 
     accounting_delta_adapter: PoSAccountingDeltaAdapter<A>,
     accounting_block_undo: AccountingBlockUndoCache,
+
+    account_nonce: BTreeMap<AccountType, Option<u128>>,
```

Comment:
If the Option in the value is used to indicate deletion (as from my understanding in the code), please use CachedOperation instead of Option.